library/pod-security-policy/host-filesystem/src.rego

```diff
@@ -0,0 +1,66 @@
+ package k8spsphostfilesystem
+
+violation[{"msg": msg, "details": {}}] {
+    volume := input_hostpath_volumes[_]
+    not input_hostpath_allowed(volume)
+    msg := sprintf("HostPath volume %v is not allowed, pod: %v. Allowed path: %v", [volume, input.review.object.metadata.name, input.parameters.allowedHostPaths])
+}
+
+input_hostpath_allowed(volume) {
+    # An empty list means there is no restriction on host paths used
+    input.parameters.allowedHostPaths == []
+}
+
+input_hostpath_allowed(volume) {
+    allowedHostPath := input.parameters.allowedHostPaths[_]
+    path_matches(allowedHostPath.pathPrefix, volume.hostPath.path)
+    not allowedHostPath.readOnly
```

Comment:
Can make this `not allowedHostPath.readOnly == true` and it'll cover both the case when the field is missing and the case when it's provided but set to false.